### DISCUSSION TITLE
No Connect When Opening Firefox Browser With Cypress. Firefox browser window is all white.

### DISCUSSION THREAD

**Discussion by thepixelhead** [cypress-io/cypress#29994] [Questions and Help]
The Firefox Browser keeps beeing completely white when opening with Cypress. Until yesterday all worked fine, since today I have this problem.

I started Cypress in the Visual Studio Code Terminal with

`npx cypress open`

The Cypress Launchpad window opens. I choosed the browser Firefox. I pressed the button "Openening E2E Testing in Firefox".

Then the Firefox browser opens ... but stays completely white.

In the Terminal is this info:

`Still waiting to connect to Firefox, retrying in 1 second (attempt 42/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 43/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 44/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 45/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 46/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 47/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 48/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 49/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 50/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 51/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 52/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 53/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 54/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 55/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 56/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 57/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 58/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 59/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 60/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 61/62)
Still waiting to connect to Firefox, retrying in 1 second (attempt 62/62)`

Expected behaviour is the Firefox browser shows a list of E2E specs, the list of all Cypress test cases.

Additional info on the browser handling: 
I opened the Firefox browser before using Cypress for other things. All fine. When starting Cypress first time today I remeber Firefox was suddenly updating automatically. When finally open all was white. All other attempts opening Firefox with Cypress in the following had all the same result:

`Still waiting to connect to Firefox, retrying in 1 second (attempt 62/62)`

The Firefox Browser keeps beeing completely white when openning with Cypress. Until yesterday all worked fine, since today I have this problem.

Versions:

- Cypress v12.14.0
- Windows 10 Enterprise 
- Firefox 129.0 (64-Bit)

Is there anyone else with the same issue? Is there a known solution to this?

**Comment by portyu9** [comment]
Hi @thepixelhead 

This is caused by the Firefox `129` change, not by your test code.

Firefox 129 changed/deprecated the CDP behavior Cypress was using. Cypress tracked this in issue `#29713`, and the fix was released in **Cypress 13.13.0**:

> Fixed an issue where Firefox 129 (Firefox Nightly) would not launch with Cypress.

You are running:

```text
Cypress: 12.14.0
Firefox: 129.0
```

so the Cypress version predates that Firefox compatibility fix.

Upgrade Cypress:

```sh
npm install --save-dev cypress@latest
```

or, if you need the smallest historical upgrade, use at least the Cypress release containing the Firefox 129 fix (`13.13.0`).

Then verify the installed binary:

```sh
npx cypress version
npx cypress verify
```

I would not try to solve this by changing your tests, clearing cookies, or modifying browser security options—the failure happens while Cypress is establishing its browser automation connection, before the tests run.

For a new/current installation, prefer a current Cypress version with a currently supported Firefox version rather than keeping Cypress 12 and pinning Firefox to an old release.

No screenshots or attachments were included in this discussion.

Please mark this answer as accepted if it helped, thank you.
